parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+	let mut response_event: Option<GetResponse> = None;
+
+	while height_to_fetch <= (tx_block_height + 10) {
+		let fetched_events = client_a.events().at(block_hash).await?.find_first::<GetResponse>()?;
+		match fetched_events {
+			Some(res_event) => {
+				response_event = Some(res_event.clone());
+				break
+			},
+			None => {
+				height_to_fetch += 1;
+				block_hash =
+					client_a.rpc().block_hash(Some(height_to_fetch.into())).await?.unwrap();
+			},
+		}
+	}
+
+	let encoded_value = match response_event.unwrap().0[0].clone() {
+		Some(value) => value,
+		None => {
+			panic!("Value not found")
+		},
+	};
+
+	Ok(encoded_value)
+}
+
+/// Configure the state machines and spawn the messaging relayer
+async fn create_clients(
+) -> Result<(SubstrateClient<Hyperbridge>, SubstrateClient<Hyperbridge>), anyhow::Error> {
+	let chain_a_config = SubstrateConfig {
+		state_machine: Kusama(2000),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9990".to_string(), // url from local-testnet zombienet config
+		max_rpc_payload_size: None,
+		signer: Some("e5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0".to_string()),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	let chain_b_config = SubstrateConfig {
+		state_machine: Kusama(2001),
+		hashing: None,
+		consensus_state_id: Some("PARA".to_string()),
+		rpc_ws: "ws://127.0.0.1:9991".to_string(),
+		max_rpc_payload_size: None,
+		signer: Some("e5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0".to_string()),
+		latest_height: None,
+		max_concurent_queries: None,
+	};
+
+	// setup state machines
+	let chain_a_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_a_config.clone()).await?;
+	let chain_b_sub_client =
+		SubstrateClient::<KeccakSubstrateChain>::new(chain_b_config.clone()).await?;
+
+	Ok((chain_a_sub_client, chain_b_sub_client))
+}
+
+/// Assertion is on ismp related events, and state changes on source and destination chain
+/// Alice in chain A sends 100_000 tokens to Alice in chain B
+#[tokio::test(flavor = "multi_thread")]
+async fn submit_transfer_function_works() -> Result<(), anyhow::Error> {
+	log_setup()?;
+	let (chain_a_sub_client, chain_b_sub_client) = create_clients().await?;
+
+	log::info!(
+		"🧊integration test for para:{} to para {}: fund transfer",
+		chain_a_sub_client.clone().state_machine_id().state_id,
+		chain_b_sub_client.clone().state_machine_id().state_id
+	);
+
+	// initiate message relaying task
+	let tx_payment = Arc::new(
+		TransactionPayment::initialize(&"/tmp/dev.db") // out of hyperbridge directory
+			.await
+			.map_err(|err| anyhow!("Error initializing database: {err:?}"))?,
+	);
+
+	let tokio_handle = tokio::runtime::Handle::current();
+	let task_manager = TaskManager::new(tokio_handle, None)?;
+	let relayer_config = RelayerConfig::default();
+
+	let (chain_a_client, chain_b_client) = (
+		Arc::new(chain_a_sub_client.clone()) as Arc<dyn IsmpProvider>,
```

Comment:
Type cast not needed
```suggestion
		Arc::new(chain_a_sub_client.clone()) as Arc<dyn IsmpProvider>,
```

Reply:
In client map the compiler complains without type casting